### PR TITLE
[RLlib] Issue 8319 DDPG (MA or num_envs_per_worker > 1) broken.

### LINKED ISSUE #8319: [rllib] Vectorization & multi-agent are broken in DDPG (both TF and Torch)
### What is the problem?

`./train.py --run=DDPG --env=MountainCarContinuous-v0 --config='{"num_envs_per_worker": 2}'`
and
`./train.py --run=DDPG --env=MountainCarContinuous-v0 --config='{"num_envs_per_worker": 2}' --torch`

both currently crash. The issue seems to be that while the input to compute_actions() is a batch of N observations, only 1 action is returned as output. I ran into this while debugging a hang in a multi-agent test case (hung due to a bug in the env triggered by vectorization returning 1 action instead of N actions).

### PR DESCRIPTION
**Pull request by sven1977**
This PR fixes a bug in `utils/explorations/random.py` where the random sample is not batched to the required batch size. This leads to various bugs in algos that rely on the Random exploration component (e.g. DDPG in its initial random phase).

Fixes issue #8319 

Closes #8319 

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/env/base_env.py [hunk 1]
```diff
@@ -445,3 +452,23 @@ def reset(self):
         self.last_infos = {agent_id: {} for agent_id in self.last_obs.keys()}
         self.last_dones["__all__"] = False
         return self.last_obs
+
+
+def unwrap_gym_env(env):
```
Comment (ericl) [inline rllib/env/base_env.py:457]: Could we keep this in a separate PR? Shouldn't mix up changes

Comment (sven1977) [reply rllib/env/base_env.py:457]: Good catch. This should not have been here.

**Review by ericl — APPROVED** [review]
